lib/rules/no-duplicate-imports.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @fileoverview Restrict usage of duplicate imports.
+ * @author Simen Bekkhus
+ * @copyright 2016 Simen Bekkhus. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/**
+ * Returns the name of the module imported or re-exported.
+ *
+ * @param {ASTNode} node - A node to get.
+ * @returns {String|Boolean} the name of the module, or false if no name.
+ */
+function getValue(node) {
+    if (node && node.source && node.source.value) {
+        return node.source.value.trim();
+    }
+
+    return false;
+}
+
+/**
+ * Checks if the name of the import or export exists in the given array, and reports if so.
+ *
+ * @param {RuleContext} context - The ESLint rule context object.
+ * @param {ASTNode} node - A node to get.
+ * @param {String} value - The name of the imported or exported module.
```

Comment:
We use the lowercase version of primitive types (`string` instead of `String`, `boolean` instead of `Boolean` etc.)

This comment applies to all you JSDocs